virttest/utils_test/libvirt.py

```diff
@@ -2371,19 +2372,47 @@ def create_tpm_dev(params):
     return tpm_dev
 
 
-def get_vm_device(vmxml, dev_tag, index=0):
+def get_vm_device(vmxml, dev_tag, dev_dict=None, index=0):
```

Comment:
Command-query separation (CQS)
A principle of imperative computer programming. It was devised by Bertrand Meyer as part of his pioneering work on the Effective programming language. It states that every method should either be a command that performs an action, or a query that returns data to the caller, but not both

As per Command-query separation (CQS) principal, don't try to mix get and add operations together in one method.

Reply:
Yes ,Thank you chunfu ,That's a great principle :smile: , 
But I think this function only implement one operation even after my updated, which is to get one device you want. it follows the principle, right ? And this updated propose is gotten from dzheng @dzhengfy Please help check

Reply:
If you want to create one method,which achieve the purpose of creating one if not present when call get_xx, I suggest you create one separate method. Don't just modify them on primitive get_vm_device.